src/ui/components/LedgerAdmin.js

```diff
@@ -98,35 +114,58 @@ export const LedgerAdmin = () => {
   };
 
   const renderIdentities = () => {
-    const renderIdentity = (i: Identity, notLastElement: boolean) => (
+    const renderIdentity = (
+      i: Identity,
+      index: number,
+      notLastElement: boolean
+    ) => (
       <>
-        <ListItem button onClick={() => setActiveIdentity(i)} key={i.id}>
+        <ListItem
+          {...getOptionProps({i, index})}
+          button
+          onClick={() => setActiveIdentity(i)}
+          key={i.id}
+        >
           {i.name}
         </ListItem>
         {notLastElement && <Divider />}
       </>
     );
     const numAccounts = ledger.accounts().length;
+    const accounts = inputValue ? groupedOptions : ledger.accounts();
     return (
       <>
-        {ledger
-          .accounts()
-          .map((a) => a.identity)
-          .map((identity, index) =>
-            renderIdentity(identity, index < numAccounts - 1)
-          )}
+        {accounts.length > 0 ? (
+          accounts
+            .map((a) => a.identity)
+            .sort(function (a, b) {
+              return a.name.localeCompare(b.name);
+            })
+            .map((identity, index) =>
+              renderIdentity(identity, index, index < numAccounts - 1)
+            )
+        ) : (
+          <ListItem>{"No Identities are here"}</ListItem>
+        )}
       </>
     );
   };
-
   return (
     <Container className={classes.root}>
       <span className={classes.centerRow}>
-        <h1 className={classes.IdentitiesHeader}>Identities</h1>{" "}
-        {ledger.accounts().length > 0 && <h3> (click one to update it)</h3>}
+        <h1 className={classes.IdentitiesHeader}>Identities</h1>
+        <TextField
+          className={classes.input}
+          label="Search"
```

Comment:
If we leave this where it is I'm inclined to call this a filter box since it doesn't really return anything. it just filters the list. 
```suggestion
          label="Filter"
```